File: src/test/java/StringToKeypadConverterTest.java
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.*;
import static org.junit.jupiter.params.provider.Arguments.arguments;

class StringToKeypadConverterTest {

    private static StringToKeypadConverter stringToKeypadConverter;

    @BeforeAll
    static void beforeAll() {
        stringToKeypadConverter = new StringToKeypadConverter();
    }

    @ParameterizedTest
    @MethodSource("testValuesProvider")
    void convertsString(String input, String result) {
        String s = stringToKeypadConverter.convertToKeyPad(input);
        assertEquals(result, s);
    }

    static Stream<Arguments> testValuesProvider() {
        return Stream.of(
                arguments("hello world", "4433555 555666096667775553"),
                arguments("foo  bar", "333666 6660 022 2777"),
                arguments("yes", "999337777"),
                arguments("hi", "44 444"),
                arguments(" abc ", "02 22 2220")
        );
    }

    @Test
    void skipsUnknownValues() {
        String s = stringToKeypadConverter.convertToKeyPad("Ay*e#s5");
        assertEquals("999337777", s);
    }
}

File: C#/AsposeTest/UnitTestProject/StringToKeypadConverterTest.cs
using AsposeTest;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

namespace AsposeUnitTestProject
{
    [TestClass]
    public class StringToKeypadConverterTest
    {
        private StringToKeypadConverter stringToKeypadConverter;

        [TestInitialize]
        public void Setup()
        {
            stringToKeypadConverter = new StringToKeypadConverter();
        }

        [TestMethod]
        [DynamicData(nameof(testValuesProvider), DynamicDataSourceType.Method)]
        public void convertsString(string input, string result)
        {
            string s = stringToKeypadConverter.convertToKeyPad(input);
            Assert.AreEqual(result, s);
        }

        static IEnumerable<object[]> testValuesProvider()
        {
            yield return new object[] { "hello world", "4433555 555666096667775553" };
            yield return new object[] { "foo  bar", "333666 6660 022 2777" };
            yield return new object[] { "yes", "999337777" };
            yield return new object[] { "hi", "44 444" };
            yield return new object[] { " abc ", "02 22 2220" };
        }

        [TestMethod]
        public void skipsUnknownValues()
        {
            string s = stringToKeypadConverter.convertToKeyPad("Ay*e#s5");
            Assert.AreEqual("999337777", s);
        }
    }
}
